packages/framework/aqueduct/src/data-object-factories/dataObjectFactory.ts

```diff
@@ -40,6 +54,7 @@ export class DataObjectFactory<
 		optionalProviders: FluidObjectSymbolProvider<I["OptionalProviders"]>,
 		registryEntries?: NamedFluidDataStoreRegistryEntries,
 		runtimeFactory: typeof FluidDataStoreRuntime = FluidDataStoreRuntime,
+		convertDataFn?: (runtime: FluidDataStoreRuntime, root: ISharedDirectory) => Promise<void>,
```

Comment:
I'm not sure I follow the usage pattern.
As a user, am I supposed to know (before deciding to supply convertDataFn or not) if the instance of data store I'm dealing with is in old or new format?
I'd think the answer should be no, and thus there should be (in addition to convertDataFn) some kind of testFn that can tell if conversion is required, i.e. if the instance of data store is in old schema or new schema.

Reply:
Maybe this is out of scope for now, please consider daisy chaining converters (i.e. assuming that we have Schema1 -> Schema2 -> Schema3 kind of conversion, with maybe 1 -> 2 optional (i.e. maybe we already deal with object in Schema2)